mlir/lib/Dialect/Func/IR/FuncOps.cpp

```diff
@@ -128,7 +128,9 @@ LogicalResult ConstantOp::verify() {
   Type type = getType();
 
   // Try to find the referenced function.
-  auto fn = (*this)->getParentOfType<ModuleOp>().lookupSymbol<FuncOp>(fnName);
+  SymbolTable symbolTable(
+      (*this)->getParentWithTrait<mlir::OpTrait::SymbolTable>());
+  auto fn = symbolTable.lookup<FuncOp>(fnName);
```

Comment:
Building a symbolTable is not cheap unfortunately, so this call is pretty expensive.

More importantly the verifier has support for SymbolTable verifications, see the last paragraph for this section: https://mlir.llvm.org/docs/SymbolsAndSymbolTables/#referencing-a-symbol

> Operations that reference a Symbol and want to perform verification and general mutation of the symbol should implement the SymbolUserOpInterface to ensure that symbol accesses are legal and efficient.

Reply:
Thanks for your feedback. I have revised the code to use `SymbolUserOpInterface` in `func::ConstantOp`, the resulting `verifySymbolUses()` eliminated the need for building a symbol table for verification.